### DISCUSSION TITLE
Window title cannot be switched off

### DISCUSSION THREAD

**Discussion by flexid2000** [lxqt/lxqt#2437] [Q&A] [answered]
##### Expected Behavior

In applications like Firefox and Thunderbird, there was an option to switch off the title bar. This option should be still available.

##### Current Behavior

Since recently, this option is gone. As a consequence, the window manager title bar is always used.

##### Possible Solution

I have no idea which module is responsible for that behaviour. In KDE, the option is still present so it cannot be dependent on firefox.

##### Steps to Reproduce (for bugs)

1. Open Firefox, choose `customize icon bar`.
2. There is no option to uncheck the title bar.

##### System Information

* Distribution & Version: Arch Linux
* Kernel: 6.3.7-arch1-1
* Qt Version: 5.15.9
* liblxqt Version: 1.3.0-1
* WM: kwin_x11 or openbox (identical behaviour)

**Comment by tsujan** [comment]
Not related to LXQt. Title-bar is controlled by the window manager.

**Comment by flexid2000** [comment]
How come you have got different results using the same window manager (kwin_x11): it displays this option in KDE but not in LXQt?

**Comment by tsujan** [comment]
I also use KWin, and

![firefox](https://github.com/lxqt/lxqt/assets/981076/722e2157-2dbf-48fb-bba0-1dd40d69c8f9)

Which KWin setting disables it in your case, I have no idea. But LXQt doesn't and can't have any effect on such things.

**Comment by flexid2000** [comment]
Mysterious. Well, I'll close the issue.

**Comment by tsujan** [comment]
OK, it isn't related to KWin either. I upgraded Firefox from V113.0.1 to V114.0.1 and the option disappeared. You might want to report it to Firefox then.

**Comment by tsujan** [comment]
> t displays this option in KDE but not in LXQt?

If it's shown under KDE, then Firefox 114.0.1 may enable it only under certain DEs, while Firefox 113.0.1 enabled it everywhere.

This is nothing compared to so many backward-incompatible changes in Firefox...

**Comment by stefonarch** [comment]
I've the titlebar alway hidden for months in firefox  (x11 and wayland), but to be honest I don't remember how I did, maybe in` about:config` and it never changed.  version is now 114.01. The buttons go into the tab bar here.
![screen_area_mar_20:29:48_](https://github.com/lxqt/lxqt/assets/10681413/897d6cd2-e266-45cb-bb4a-3fe516aacdf9)

**Comment by stefonarch** [comment]
I see the option, right click on the tool bar, on the bottom left is a checkbox.

**Comment by tsujan** [comment]
> I see the option, right click on the tool bar, on the bottom left is a checkbox.

That's the option in my screenshot. It disappeared as soon as I upgraded to 114.0.1. It may be there in your case because you had enabled it before the upgrade. A mess in Firefox.

**Comment by tsujan** [comment, marked as answer]
Problem solved. Firefox has changed its behavior such that if the EV `GTK_CSD` is set to 0, the option in the screenshot won't be shown anymore. That's a bad change because if `GTK_CSD` is set to 1, other GTK apps will follow it.

You can change `GTK_CSD` to 1 in LXQt Session Settings if you can tolerate GTK apps with GTK titlebars.

I'm changing this to a "discussion"; it might be useful to other users.

**Comment by stefonarch** [comment]
O, I missed your screenshot.
 I've a second profile with titlebar, and the option is still sitting there. 

EDIT: hm, I've `GTK_CSD=0` set so this looks more complicated.

**Reply by tsujan** [reply]
> EDIT: hm, I've GTK_CSD=0 set so this looks more complicated.

Are you in an X11 session of LXQt? If you're in Wayland, everything is different.

**Reply by stefonarch** [reply]
I'm always in wayland nowadays. will check now.

**Reply by stefonarch** [reply]
Yes, the option is gone under X11 but still no titlebar :)

So it looks like if someone prefers disabling GTK decoration with `GTK_CSD=0` but wants no titlebar he has to login once in wayland, start firefox and disable the titlebar.

**Reply by tsujan** [reply]
> So it looks like if someone prefers...

In short, it's a mess ;)

I also upgraded Chromium. Its titlebar option is intact, regardless of `GTK_CSD`.

**Comment by yan12125** [comment]
Bug report for Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1827563
